examples/pytorch/language-modeling/run_clm.py

```diff
@@ -506,13 +506,28 @@ def main():
             print('> Sharding tensor', name, param.shape)
             tensor = model_args.spmd_tensor_sharding
             fsdp = num_devices // tensor
-            assert fsdp * tensor == num_devices
-            mesh = xs.Mesh(device_ids, (fsdp, tensor))
+
+            col_mesh = xs.HybridMesh(ici_mesh_shape=(fsdp, tensor))
+            row_mesh = xs.HybridMesh(ici_mesh_shape=(tensor, fsdp))
             if len(param.shape) == 1:
-                xs.mark_sharding(param, mesh, (1,))
+                # Mark sharding on FSDP axis to replicate in tensor dimension
+                xs.mark_sharding(param, col_mesh, (0,))
```

Comment:
@alanwaketan This is currently broken, but I wanted to get your input. Should we shard 1D tensors FSDP-mode instead of sharding on the tensor parallel dimension?

Reply:
I think it's probably better to shard it according to the tensor dimension. I guess you have two options here:
1. Don't use HybridMesh.
2. Or Shard it on the FSDP dimension.

Probably there aren't any differences.

Reply:
I was going to try using row_mesh instead of col_mesh, which would let me do `xs.mark_sharding(param, row_mesh, (1,))` to shard on the FSDP dimension.

Reply:
cc @Liyang90 if you have any thoughts